src/include/ipc/topology.h

```diff
@@ -86,12 +86,12 @@ struct sof_ipc_comp {
 /*
  * overrun will cause ring buffer overwrite, instead of XRUN.
  */
```

Comment:
It is not clear from the commit message why we change the bit positions? What's wrong with the current positions (0 and 1?)

Reply:
I think the intent is to separate memory specific parameters from those specific to the buffer itself. For example, bits 0-15 could contain memory caps like allocating the buffer from a DMA-able region while 16-31 would contain stuff that matters for topology (overrun/underrun permitted).

Reply:
yes, we added "flags" field which was completely unnecessary as we have about 24 bits left in "caps". So the bit position 0 and 1 we're in the other parameter before. If we can use the caps for other parameters (so exactly like the upper 16bit and leave still 8 for possible future mem flags), I don't have to touch topology. Adding the new flags to topology would touch every buffer definition in every topology and I'm paranoid with this kind of changes, as we have screwed up tplg+fw+kernel combinations before....

Reply:
So this is a very good example why we need always driver side PR together with fw PR to see the possible topology and other changes before we merge firmware.

Reply:
@paulstelian97 @juimonen thanks for explanation. Can you please add this explanation in the commit message.

Reply:
@juimonen do we also need to bump the tplg abi?

Reply:
@dbaluta hmm I think it should be visible in the commit messages... there's like 5 commits? the PR message might not contain all relevant info. Sorry if I'm wrong here, I'll correct it then... I'm not including my own rants how stupid we sometimes do things in commit messages :)